inspectkv/inspectkv.go

```diff
@@ -108,20 +123,122 @@ func ScanIndexData(txn kv.Transaction, kvIndex kv.Index, startVals []interface{}
 	return idxRows, nextVals, nil
 }
 
-// RecordData is the record data composed of a handle and column values.
-type RecordData struct {
-	Handle int64
-	Values []interface{}
+// ScanIndexColData scans the index handles and values in a limited number, according to the corresponding column.
+// It returns data and the next startHandle until it doesn't have data, then returns data is nil and
+// the next startHandle is the handle which can't get data.
+// If limit = -1, it returns the index data of the whole.
+func ScanIndexColData(retriever kv.Retriever, t table.Table, idx *column.IndexedCol, startHandle, limit int64) (
+	[]*RecordData, int64, error) {
+	cols := make([]*column.Col, len(idx.Columns))
+	for i, col := range idx.Columns {
+		cols[i] = t.Cols()[col.Offset]
+	}
+
+	return scanTableData(retriever, t, cols, startHandle, limit)
 }
 
-// ScanTableData scans table row handles and column values in a limited number.
-// It returns data and the next startKey.
-func ScanTableData(t table.Table, retriever kv.Retriever, startHandle, limit int64) (
+// EqualRecords returns ture if a is equal to b.
+func EqualRecords(a, b []*RecordData) bool {
```

Comment:
You can returns the index `i` when encountered not equall record pair, so we know which pair od record is different.